tox.ini

```diff
@@ -1,15 +1,25 @@
 [tox]
-envlist = py3, flake8, pycodestyle
+envlist = py3, flake8, flake8-{trusty,xenial,bionic}
+
+# N.B. If this needs to be expanded to include a third type of test, a
+# "new-test: new-test-commands" line should be added to the commands stanza
+# (and perhaps the deps stanza, if additional dependencies are required), and
+# the default "!flake8" line should be extended to "!flake8-!new-test"
 
 [testenv]
 deps =
     -rrequirements.txt
     -rtest-requirements.txt
-commands = nose2 --with-coverage --coverage uaclient/ uaclient
-
-[testenv:flake8]
-commands = flake8 uaclient setup.py
-
-[testenv:pycodestyle]
-commands = pycodestyle {posargs:uaclient setup.py}
+    trusty: -cdev/trusty-constraints.txt
+    xenial: -cdev/xenial-constraints.txt
+    bionic: -cdev/bionic-constraints.txt
+commands =
+    !flake8: nose2 --with-coverage --coverage uaclient/ uaclient
+    flake8: flake8 uaclient setup.py
 
+[flake8]
+ignore =
+    # Older versions of flake8 et al don't permit the now-recommended-by-PEP-8
+    # parameter spacing for annotated function arguments with defaults (e.g.
+    # `def spam(ham: str = "eggs"):`).
+    E251,
```

Comment:
In debian/rules we call nose and flake8 directly instead of running it via tox, so we use the in-distro versions of packages. That means this ignore statement will not have any effect when building the package.

Reply:
Per [the flake8 documentation](http://flake8.pycqa.org/en/3.1.1/user/configuration.html#configuration-locations), flake8 will look in `setup.cfg`, `tox.ini` or `.flake8` for its configuration (regardless of how its being run). Given that we already have a `tox.ini`, I opted for that.

I've also just confirmed that this will work in trusty (using a container):

```
# apt install python3-flake8
...
# cat << EOF > foo.py
> def foo(x = "y"):
>     pass
> EOF
# flake8 foo.py 
foo.py:1:10: E251 unexpected spaces around keyword / parameter equals
foo.py:1:12: E251 unexpected spaces around keyword / parameter equals
# cat << EOF > tox.ini
> [flake8]
> ignore =
>     E251,
> EOF
# cat << EOF > tox.ini
> [flake8]
> ignore =
>     # This is a comment
>     E251,
> EOF
# flake8 foo.py 
# rm tox.ini
# flake8 foo.py 
foo.py:1:10: E251 unexpected spaces around keyword / parameter equals
foo.py:1:12: E251 unexpected spaces around keyword / parameter equals
```

I'm happy to switch to one of the other two if you think that would be preferable.

Reply:
TIL... thanks for that!